Climate_police/homework 7/temperature.py

```diff
@@ -0,0 +1,178 @@
+        autocolorscale = True,
+        reversescale = False,
+        marker = dict(
+            line = dict (
+                color = 'rgb(180,180,180)',
+                width = 0.5
+            ) ),
+        colorbar = dict(
+            autotick = False,
+            title = 'Average land temperature'),
+      ) ]
+
+    layout = dict(
+        title = 'Average land temperature by country',
+        geo = dict(
+            showframe = False,
+            showcoastlines = False,
+            showocean = True,
+            oceancolor = 'rgb(10,200,255)',
+            projection = dict(
+                type = 'Mercator'
+                )
+            )
+        )
+
+    fig = dict( data=data, layout=layout )
+    py.iplot( fig, validate=False, filename='d3-world-map' )
+
+def globe_graph():
+    """
+    Cleaning data & plotting average land temperature
+    """
+    global_temp_country_clean = global_temp_country[~global_temp_country['Country'].isin(
+    ['Denmark', 'Antarctica', 'France', 'Europe', 'Netherlands',
+     'United Kingdom', 'Africa', 'South America'])]
+
+    global_temp_country_clean = global_temp_country_clean.replace(
+       ['Denmark (Europe)', 'France (Europe)', 'Netherlands (Europe)', 'United Kingdom (Europe)'],
+       ['Denmark', 'France', 'Netherlands', 'United Kingdom'])
+
+    #Let's average temperature for each country
+
+    countries = np.unique(global_temp_country_clean['Country'])
+    mean_temp = []
+    for country in countries:
+        mean_temp.append(global_temp_country_clean[global_temp_country_clean['Country'] ==
+                                                   country]['AverageTemperature'].mean())
+
+
+
+    data = [ dict(
+            type = 'choropleth',
+            locations = countries,
+            z = mean_temp,
+            locationmode = 'country names',
+            text = countries,
+            marker = dict(
+                line = dict(color = 'rgb(0,0,0)', width = 1)),
+                colorbar = dict(autotick = True, tickprefix = '',
+                title = '# Average\nTemperature,\n°C')
+                )
+           ]
+
+    layout = dict(
+        title = 'Average land temperature in countries',
+        geo = dict(
+            showframe = False,
+            showocean = True,
+            oceancolor = 'rgb(0,255,255)',
+            projection = dict(
+            type = 'orthographic',
+                rotation = dict(
+                        lon = 60,
+                        lat = 10),
+            ),
+            lonaxis =  dict(
+                    showgrid = True,
+                    gridcolor = 'rgb(102, 102, 102)'
+                ),
+            lataxis = dict(
+                    showgrid = True,
+                    gridcolor = 'rgb(102, 102, 102)'
+                    )
+                ),
+            )
+
+    fig = dict(data=data, layout=layout)
+    py.iplot(fig, validate=False, filename='worldmap')
+
+
+def remove_data():
```

Comment:
I would delete also pycache files, adding the related function in the remove_data() function.